runtime-management/runtime-management-impl/src/main/java/io/cloudslang/runtime/impl/python/external/ExternalPythonExecutor.java

```diff
@@ -190,7 +213,30 @@ private PythonEvaluationResult runPythonEvalProcess(String pythonPath, String pa
         ProcessBuilder processBuilder = preparePythonProcess(executionEnvironment, pythonPath);
 
         try {
-            String returnResult = getResult(payload, processBuilder);
+            String returnResult = getResult(payload, processBuilder, EVALUATION_TIMEOUT);
+
+            EvaluationResults scriptResults = objectMapper.readValue(returnResult, EvaluationResults.class);
+            String exception = scriptResults.getException();
+            if (!StringUtils.isEmpty(exception)) {
+                logger.error(String.format("Failed to execute script {%s}", exception));
+                throw new ExternalPythonEvalException(exception);
+            }
+            context.put("accessed_resources_set", (Serializable) scriptResults.getAccessedResources());
+            //noinspection unchecked
+            return new PythonEvaluationResult(processReturnResult(scriptResults), context);
+        } catch (IOException | InterruptedException | ExecutionException | TimeoutException e) {
+            logger.error("Failed to run script. ", e.getCause() != null ? e.getCause() : e);
+            throw new RuntimeException("Failed to run script.");
+        }
+    }
```

Comment:
All this code si the same in the test method, except for the timeout value and the exception handling. Can you do one method for both and just leverage between eval/test ? Maybe just send the timeout value and you can throw the same timeout exception message for eval as well.

Reply:
done